### PR TITLE
DM-33779: Implement drp_pipe pipeline unit tests

### PR DESCRIPTION
**Pull request by leeskelvin**
## Checklist

- [x] ran Jenkins
- [x] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: python/lsst/pipe/base/tests/pipelineStepTester.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    the named pipeline, either as intermediates or as final outputs.
+
+    These sets will be checked against user-supplied sets to ensure that the
+    named pipeline may still be run without raising a missing data error.
+
+    Parameters
+    ----------
+    filename : `str`
+        The full path to the pipeline YAML
+    step_suffixes : `list` of `str`
```
Comment (arunkannawadi) [inline python/lsst/pipe/base/tests/pipelineStepTester.py:56]: For documentation of containers, you should use this style:
https://developer.lsst.io/python/numpydoc.html#sequence-types

Comment (arunkannawadi) [reply python/lsst/pipe/base/tests/pipelineStepTester.py:56]: Why are these lists, as opposed to say sets? Does the order matter here?

**Review by arunkannawadi — APPROVED** [review]
Minor inline comments on documentation, otherwise all LGTM!

File: python/lsst/pipe/base/tests/pipelineStepTester.py (hunk 1 above)
Comment (leeskelvin) [reply python/lsst/pipe/base/tests/pipelineStepTester.py]: Documentation updated as per the referenced style guide.

Comment (leeskelvin) [reply python/lsst/pipe/base/tests/pipelineStepTester.py]: Yes, the order is important in `step_suffixes`. Specifically, the pipeline needs to be progressively built in the order in which it's executed. This allows us to build up a list of overall expected input dataset types, and known expected outputs.

The `initial_dataset_types` parameter needs to be a list as it contains unhashable types.

**Pull request merged**